docs/guides/app-payments/build-on-mobile.mdx

```diff
@@ -21,46 +19,199 @@ This page details how MobilePay app payments should be implemented in your Andro
+          data = Uri.parse(deepLink)
+      }
+      mobilePayContract.launch(intent)
+  } catch (exception: ActivityNotFoundException) {
+      // Inform the user that MobilePay is not installed.
+  }
+}
+
+companion object {
+    private const val SUCCESS_RESULT_CODE = Activity.RESULT_OK
+    private const val REJECT_RESULT_CODE = Activity.RESULT_CANCELED
+    private const val FAILED_RESULT_CODE = 1
+    private const val UNKNOWN_RESULT_CODE = 2
+}
+```
+
 </TabItem>
 
-<TabItem value="ios">
+<TabItem value="activity_result_kotlin">
 
-This guide assumes:
+```kotlin title="Kotlin"
+// Use the link received from backend to launch MobilePay.
+fun openMobilePay(deepLink: String) {
+    try {
+        val intent = Intent(Intent.ACTION_VIEW).apply {
+            data = Uri.parse(deepLink)
+        }
+        startActivityForResult(intent, MOBILEPAY_PAYMENT_REQUEST_CODE)
+    } catch (exception: ActivityNotFoundException) {
+        // Inform the user that MobilePay is not installed.
+    }
+}
 
-- You already have backend services integrated with [MobilePay APIs](/docs/guides/app-payments/how-it-works).
-- Your app handles a custom URL scheme. See [Apple documentation](https://developer.apple.com/documentation/xcode/defining-a-custom-url-scheme-for-your-app) for more details.
-- You are generally familiar with developing applications on iOS.
+override fun onActivityResult(
+    requestCode: Int,
+    resultCode: Int,
+    data: Intent?
+) {
+  super.onActivityResult(requestCode, resultCode, data)
+  if (requestCode == MOBILEPAY_PAYMENT_REQUEST_CODE) {
+    when (resultCode) {
+      SUCCESS_RESULT_CODE -> {}
+      REJECT_RESULT_CODE -> {}
+      FAILED_RESULT_CODE -> {}
+      UNKNOWN_RESULT_CODE -> {}
+    }
+  }
+}
+
+companion object {
+  const val MOBILEPAY_PAYMENT_REQUEST_CODE = 1337 // Any request code you choose
+
+  private const val SUCCESS_RESULT_CODE = Activity.RESULT_OK
+  private const val REJECT_RESULT_CODE = Activity.RESULT_CANCELED
+  private const val FAILED_RESULT_CODE = 1
+  private const val UNKNOWN_RESULT_CODE = 2
+}
+```
 
 </TabItem>
+
+<TabItem value="java">
+
+```java title="Java"
+private static final int MOBILEPAY_PAYMENT_REQUEST_CODE = 1337; // Any request code you choose
+
+private static final int SUCCESS_RESULT_CODE = Activity.RESULT_OK;
+private static final int REJECT_RESULT_CODE = Activity.RESULT_CANCELED;
+private static final int FAILED_RESULT_CODE = 1;
+private static final int UNKNOWN_RESULT_CODE = 2;
+
+// Use the link received from backend to launch MobilePay.
+void openMobilePay(String deepLink) {
+    try {
+        Intent intent = new Intent(Intent.ACTION_VIEW);
+        intent.setData(Uri.parse(deepLink));
+        startActivityForResult(intent, MOBILEPAY_PAYMENT_REQUEST_CODE);
+    } catch (android.content.ActivityNotFoundException exception) {
+        // Inform the user that MobilePay is not installed.
+    } catch (NullPointerException exception) {
```

Comment:
Couldn't we just annotate `deepLink` with NonNull?

Reply:
I think that annotation requires a dependancy, so I wanted to avoid that.